libs/dh/shared/feature-highlight/src/dh-shared-highlight.worker.ts

```diff
@@ -19,13 +19,16 @@
 import hljs from 'highlight.js/lib/core';
 import json from 'highlight.js/lib/languages/json';
 import xml from 'highlight.js/lib/languages/xml';
+import type { Input } from './types';
 
 hljs.registerLanguage('json', json);
 hljs.registerLanguage('xml', xml);
 
-onmessage = (event) => {
-  const { data } = event;
+onmessage = (event: MessageEvent<Input>) => {
+  const {
+    data: { language, data },
+  } = event;
   if (!data) return postMessage('');
-  const result = hljs.highlightAuto(event.data, ['xml', 'json']);
+  const result = hljs.highlight(data, { language });
```

Comment:
issue: in messages, the documents can be either xml or json and currently we do not know the formats beforehand. so if we want to move away from the auto detection, then we first need to implement something (in the BFF perhaphs) that detects the format and passes it along to the client.

Reply:
I made it fallback to auto